cql3/describe_processor.cc

```diff
@@ -0,0 +1,241 @@
+    return {
+        make_lw_shared<column_specification>(KS, CF, ::make_shared<column_identifier>("keyspace_name", true), utf8_type),
+        make_lw_shared<column_specification>(KS, CF, ::make_shared<column_identifier>("type", true), utf8_type),
+        make_lw_shared<column_specification>(KS, CF, ::make_shared<column_identifier>("name", true), utf8_type)
+    };
+}
+
+static std::vector<lw_shared_ptr<column_specification>> element_metadata() {
+    auto metadata = list_metadata();
+    metadata.push_back(make_lw_shared<column_specification>(KS, CF, ::make_shared<column_identifier>("create_statement", true), utf8_type));
+    return metadata;
+}
+
+static seastar::sstring get_element_type(const schema_element& e) {
+    return std::visit(overloaded_functor {
+        [] (const keyspace_element& ks) {
+            return "keyspace";
+        },
+        [] (const schema_ptr& s) {
+            return "table";
+        },
+        [] (const std::pair<schema_ptr, index_metadata>& i) {
+            return "index";
+        }
+    }, e);
+}
+
+static seastar::sstring get_element_keyspace(const schema_element& e) {
+    return std::visit(overloaded_functor {
+        [&] (const keyspace_element& ks) -> seastar::sstring {
+            return ks->name();
+        },
+        [&] (const schema_ptr& s) -> seastar::sstring {
+            return s->ks_name();
+        },
+        [] (const std::pair<schema_ptr, index_metadata>& i) {
+            return i.first->ks_name();
+        }
+    }, e);
+}
+
+static seastar::sstring get_element_name(const schema_element& e) {
+    return std::visit(overloaded_functor {
+        [&] (const keyspace_element& ks) -> seastar::sstring {
+            return ks->name();
+        },
+        [&] (const schema_ptr& s) -> seastar::sstring {
+            return s->cf_name();
+        },
+        [] (const std::pair<schema_ptr, index_metadata>& i) {
+            return i.second.name();
+        }
+    }, e);
+}
+
+//TODO: finish implementation
+static seastar::sstring get_element_string(const schema_element& e, bool internals, bool if_not_exists) {
+    std::stringstream out;
+    std::visit(overloaded_functor {
+        [&] (const keyspace_element& ks) {
+
+        },
+        [&] (const schema_ptr& s) {
+
+        },
+        [] (const std::pair<schema_ptr, index_metadata>& i) {
+
+        }
+    }, e);
+    return out.str();
+}
+
+static keyspace_element get_keyspace(const seastar::sstring& keyspace, const std::vector<keyspace_element>& keyspaces) {
+    for (auto& ks : keyspaces) {
+        if (ks->name() == keyspace) {
+            return ks;
+        }
+    }
+    throw exceptions::invalid_request_exception(format("'{}' not found in keyspaces", keyspace));
+}
+
+static schema_ptr get_table(const seastar::sstring& table, const keyspace_element& ks) {
+    for (auto& t : ks->tables()) {
+        if (t->cf_name() == table) {
+            return t;
+        }
+    }
```

Comment:
I don't understand yet where this code fits in, but why do we need this new and slow (O(N)) algorithm to find a table in a keyspace? Don't we already have an O(1) lookup for table names?

Reply:
I've changed the implementation and this part doesn't exist anymore